### ISSUE TITLE
Custom pin definitions

### ISSUE THREAD

**Issue by Fabrizio3** [Klipper3d/klipper#5329]
so, im having some trouble with my printer, i want to use the MOSI pin on my board as a Z probe input (just a microswitch) as there arent any spare pins on the anet board, but i've scoured around the config and found perhaps the MCU commands is what im looking for (maybe?), specifically this: "config_endstop oid=%c pin=%c pull_up=%c stepper_count=%c", that would, in my mind, let me define any pin i wanted, as an input, with an internal pull-up (if available) and to halt the Z axis just like you need for a probe, am i going about this the wrong way? do i just put "PB5" under the probe's pin on the config file and klipper will handle the rest?     also would like to change the diag LED pin to be a pwm out for a servo (need a servo for a special toolhead)

in short: for defining custom pins (specifying what they do and the pullups), do you need to change them somewhere else? or just assign the pin in the regular config file and klipper will handle the rest?

**Comment by Fabrizio3** [comment]
apparently the probe works out of the gate, klipper is really smart huh, i'll need to test the servo just to be sure
